GVFS/GVFS.Service/Mac/CurrentUser.cs

```diff
@@ -0,0 +1,59 @@
+﻿using GVFS.Common.Tracing;
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+namespace GVFS.Service
+{
+    // TODO Mac - this is just a place holder class for the time being.
+    // Need to figure out appropriate pattern on Mac.
+
+    public class CurrentUser : IDisposable
+    {
+        private const int TokenPrimary = 1;
+        private const int StartInfoUseStdHandles = 0x00000100;
+        private const uint HandleFlagInherit = 1;
+
+        private readonly ITracer tracer;
+        private readonly IntPtr token;
+
+        public CurrentUser(ITracer tracer, int sessionId)
+        {
+            this.tracer = tracer;
+            this.token = GetCurrentUserToken(this.tracer, sessionId);
+            this.Identity = null;
+        }
+
+        public UserIdentity Identity { get; }
+
+        public bool RunAs(string processName, string args)

```

Comment:
Some of these methods (like RunAs) seem like they're only going to be relevant on Windows. If we want to start with a skeleton for now, should we just make these classes empty rather than putting in stub code now, only to delete it in the next PR?

Reply:
Removed all the un-used Handler classes and the stubs they needed. 

Looks much cleaner now -

<img width="234" alt="Screen Shot 2019-03-27 at 3 26 11 PM" src="https://user-images.githubusercontent.com/378580/55106351-1560c480-50a5-11e9-827b-b0d4348cc643.png">